sv.c

```diff
@@ -4205,6 +4205,28 @@ S_sv_buf_to_rw(pTHX_ SV *sv)
       && len                    /* and really is a string */               \
     )
 
+/* The test in this macro was also extracted from Perl_sv_setsv_flags so
+ * that it could be used elsewhere. */
+#ifdef PERL_COPY_ON_WRITE
+#define S_SvPV_shared_hkey_or_CoWable(ssv, dsv, sflags, cur, len)          \
+        (sflags & SVf_IsCOW                                                \
+            ? (!len ||                                                     \
+              ( (CHECK_COWBUF_THRESHOLD(cur,len) || SvLEN(dsv) < cur+1)    \
+                  /* If this is a regular (non-hek) COW, only so */        \
+                  /*  many COW "copies" are possible. */                   \
+                && CowREFCNT(ssv) != SV_COW_REFCNT_MAX  ))                 \
+            : ( (sflags & CAN_COW_MASK) == CAN_COW_FLAGS                   \
+                && !(SvFLAGS(dsv) & SVf_BREAK)                             \
+                && CHECK_COW_THRESHOLD(cur,len) && cur+1 < len             \
+                && (CHECK_COWBUF_THRESHOLD(cur,len) || SvLEN(dsv) < cur+1) \
+        ))
+#else
+#define S_SvPV_shared_hkey_or_CoWable(ssv, dsv, sflags, cur, len)          \
+        ( sflags & SVf_IsCOW                                               \
+        && !(SvFLAGS(dsv) & SVf_BREAK)                                     \
```

Comment:
this doesnt look like the usual macro combo to detect `SVPV HEK*`s

Reply:
It's straight out of `Perl_sv_setsv_flags`.

Reply:
I guess it is correct, testing `fCOW` by itself, in isolation, confirms its 1 of 3 things, COW STATIC, COW 255 or COW HEK.

2 permutations of "dont `Safefree()` it" are ignored, SvLEN 0 and OOK_on. But we dont care about those 2 permutation at this particular LOC.